src/plugins/integrations/MySQL2Integration.ts

```diff
@@ -40,68 +42,73 @@ class MySQL2Integration implements Integration {
-        });
-
-        span.addTags({
-          [SpanTags.SPAN_TYPE]: SpanTypes.RDB,
-          [DBTags.DB_INSTANCE]: this.config.database,
-          [DBTags.DB_USER]: this.config.user,
-          [DBTags.DB_HOST]: this.config.host,
-          [DBTags.DB_PORT]: this.config.port,
-          [DBTags.DB_TYPE]: DBTypes.MYSQL,
-          [SpanTags.TOPOLOGY_VERTEX]: true,
-          [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
-          [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
-          [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
-        });
-
-        const sequence = query.call(this, sql, values, cb);
-
-        const statement = sequence.sql;
-
-        if (statement) {
-          const statementType = statement.split(' ')[0].toUpperCase();
+        try {
+          const tracer = ThundraTracer.getInstance();
+          const parentSpan = tracer.getActiveSpan();
+
+          const span = tracer._startSpan(this.config.database, {
+            childOf: parentSpan,
+            domainName: DomainNames.DB,
+            className: ClassNames.RDB,
+            disableActiveStart: true,
+          });
+
           span.addTags({
-            [DBTags.DB_STATEMENT_TYPE]: statementType,
-            [DBTags.DB_STATEMENT]: statement,
-            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
+            [SpanTags.SPAN_TYPE]: SpanTypes.RDB,
+            [DBTags.DB_INSTANCE]: this.config.database,
+            [DBTags.DB_USER]: this.config.user,
+            [DBTags.DB_HOST]: this.config.host,
+            [DBTags.DB_PORT]: this.config.port,
+            [DBTags.DB_TYPE]: DBTypes.MYSQL,
+            [SpanTags.TOPOLOGY_VERTEX]: true,
+            [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
+            [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
+            [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
           });
-        }
 
-        const originalCallback = sequence.onResult;
+          const sequence = query.call(this, sql, values, cb);
 
-        const wrappedCallback = (err: any, res: any) => {
-          if (err) {
-            const parseError = Utils.parseError(err);
-            span.setTag('error', true);
-            span.setTag('error.kind', parseError.errorType);
-            span.setTag('error.message', parseError.errorMessage);
-            span.setTag('error.stack', parseError.stack);
-            span.setTag('error.code', parseError.code);
+          const statement = sequence.sql;
+
+          if (statement) {
+            const statementType = statement.split(' ')[0].toUpperCase();
+            span.addTags({
+              [DBTags.DB_STATEMENT_TYPE]: statementType,
+              [DBTags.DB_STATEMENT]: statement,
+              [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
```

Comment:
we might return empty operation type instead of `READ` if the operation is not known

Reply:
Done